crates/bevy_sprite/src/tilemap_chunk/tilemap_chunk_material.rs

```diff
@@ -0,0 +1,53 @@
+use crate::{Material2d, Material2dPlugin};
+use bevy_app::{App, Plugin};
+use bevy_asset::{load_internal_asset, weak_handle, Asset, Handle};
+use bevy_image::Image;
+use bevy_math::UVec2;
+use bevy_reflect::prelude::*;
+use bevy_render::render_resource::*;
+
+pub const TILEMAP_CHUNK_MATERIAL_SHADER_HANDLE: Handle<Shader> =
+    weak_handle!("40f33e62-82f8-4578-b3fa-f22989e7c4bb");
+
+/// Plugin that adds support for tilemap chunk materials.
+#[derive(Default)]
+pub struct TilemapChunkMaterialPlugin;
+
+impl Plugin for TilemapChunkMaterialPlugin {
+    fn build(&self, app: &mut App) {
+        load_internal_asset!(
+            app,
+            TILEMAP_CHUNK_MATERIAL_SHADER_HANDLE,
+            "tilemap_chunk_material.wgsl",
+            Shader::from_wgsl
+        );
+
+        app.add_plugins(Material2dPlugin::<TilemapChunkMaterial>::default());
+    }
+}
+
+/// Material used for rendering tilemap chunks.
+///
+/// This material is used internally by the tilemap system to render chunks of tiles
+/// efficiently using a single draw call per chunk.
+#[derive(Asset, TypePath, AsBindGroup, Debug, Clone)]
+pub struct TilemapChunkMaterial {
+    #[uniform(0)]
+    pub chunk_size: UVec2,
+
+    #[uniform(0)]
+    pub tile_size: UVec2,
```

Comment:
Same as above. Since the tiles are in an array texture, you can get the size with `textureDimensions`. If you wind up removing both, I think the entire tilemap chunk could just be the tileset image and the indices image.

Reply:
I removed both from the material, but kept `chunk_size` and `tile_display_size` in `TilemapChunk` as what determines the mesh size.